contracts/Insurance.sol

```diff
@@ -79,6 +80,7 @@ contract Insurance is IInsurance, Ownable, SafetyWithdraw {
     function withdraw(uint256 amount) external override {
         require(amount > 0, "INS: amount <= 0");
 
+        this.updatePoolAmount();
```

Comment:
If we are calling this internally (which we are) it makes sense to make the function public rather than external and avoid using `this.updatePoolAmount()`. This is because using the external function and calling `this.X()` makes a CALL which is more expensive.

See https://ethereum.stackexchange.com/questions/19380/external-vs-public-best-practices